src/dune/alias.ml

```diff
@@ -1,6 +1,56 @@
 open! Stdune
 open Import
 
+module Name = struct
+  type t = string
+
+  let of_string_opt s =
+    if Filename.basename s <> s then
```

Comment:
This has surprising behavior for ".", "..", "/". I think we should at least disallow the slash, but probably also the other two.

Reply:
I think we might need 2 validation functions then. I just took this validation function directly from the existing decoding function of alias names. Do you think it's worth it?

Reply:
I think it's worth having the check, yes. I don't understand why we'd need 2 validation functions though.

I don't think it's worth delaying this PR over this if you think it's non-trivial though.

Reply:
> I don't think it's worth delaying this PR over this if you think it's non-trivial though.

Otherwise I might break existing code. The current validation is the one that is currently used when parsing alias names in dune files.

Let me give it a try though. I think we can get away with making it stricter for 2.0

Reply:
You mean existing dune files that use those alias names? I didn't even consider that those could exist. :-)

Reply:
Okay, the latest commit attempts to make it more strict when possible

Reply:
You seem to special-case `[""; "."]` which is pretty far from `["."; ".."; "/"]` I suggested. What gives?

Reply:
To be clear, this is what I see:

```
Filename.basename ".";;
- : string = "."                                                                                                                                                                              Filename.basename "..";;
- : string = ".."                                                                                                                                                                             Filename.basename "/";;
- : string = "/"                                                                                                                                                                              Filename.basename "";;
- : string = "." 
```

Reply:
Yeah, that's a silly omission. I added it.